client_code/services/storage.py

```diff
@@ -106,20 +107,37 @@ class OnChainStore:
     def __init__(self):
         self._contract = None
 
+        if not _ethereum_available:
+            raise ValueError("No connected wallet found")
+        else:
+            self.provider = _ethers.providers.Web3Provider(ethereum)
+            self.signer = self.provider.getSigner()
+
     @property
     def contract(self):
         if not _ethereum_available:
             raise ValueError("No connected wallet found")
 
         if not self._contract:
-            provider = _ethers.providers.Web3Provider(ethereum)
-            signer = provider.getSigner()
-            self._contract = _ethers.Contract(self.address, self.abi, signer)
+            self._contract = _ethers.Contract(self.address, self.abi, self.signer)
 
         return self._contract
 
     def register_ballot(self, ballot):
         self.contract.createBallot(**ballot.__dict__)
 
     def cast_vote(self, ballot, ciphertext):
-        self.contract.vote(ballot.uuid, ciphertext)
+        # self.contract.vote(ballot.uuid, ciphertext)
+        from anvil.js.window import Bundlr
```

Comment:
I have a suspicion this could be done at the top of the module. I think it's loaded lazily.
Also, could we use the same syntax as we have for `_ethers`? (e.g. `_bundlr = anvil.js.import_from("Bundlr")`)